shipyard_blueprints/salesforce/shipyard_salesforce/cli/upload.py

```diff
@@ -0,0 +1,157 @@
+import os
+import sys
+import argparse
+import pandas
+
+from shipyard_salesforce import SalesforceClient
+from shipyard_templates import ExitCodeException
+from shipyard_bp_utils.files import find_files_by_regex_or_exact_match
+
+
+def get_mock_args():
+    from dotenv import load_dotenv
+
+    load_dotenv()
+    return argparse.Namespace(
+        access_token=os.getenv("SALESFORCE_ACCESS_TOKEN"),
+        domain=os.getenv("SALESFORCE_DOMAIN"),
+        username=os.getenv("SALESFORCE_USERNAME"),
+        password=os.getenv("SALESFORCE_PASSWORD"),
+        security_token=os.getenv("SALESFORCE_SECURITY_TOKEN"),
+        consumer_key=os.getenv("SALESFORCE_CONSUMER_KEY"),
+        consumer_secret=os.getenv("SALESFORCE_CONSUMER_SECRET"),
+        object_type="Account",
+        source_match_type="exact_match",
+        source_folder_name=".",
+        source_file_name="test.csv",
+        import_operation="update",
+        id_field="Id",
+    )
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--access-token", dest="access_token", required=False)
+    parser.add_argument("--consumer-key", dest="consumer_key", required=False)
+    parser.add_argument("--consumer-secret", dest="consumer_secret", required=False)
+    parser.add_argument("--username", dest="username", required=False)
```

Comment:
I think we should have some sort of validation between auth methods, it ensures that you are going the key-secret route, then those must be required in tandem.

Reply:
If i am understanding you correctly there is already a function in the salesforce utils file that handles validating auth method.

if access token is token it moves forward with that as the preference otherwise it will raise an error if all other required fields are not filled

function name: validate_client_init